pkg/scheduler/core/generic_scheduler.go

```diff
@@ -653,7 +653,7 @@ func podFitsOnNode(
 						klog.V(5).Infoln("since alwaysCheckAllPredicates has not been set, the predicate " +
 							"evaluation is short circuited and there are chances " +
 							"of other predicates failing as well.")
-						break
+						return len(failedPredicates) == 0, failedPredicates, nil
```

Comment:
The 2nd run (if appropriate) is still necessary. The reason has been explained in L609~L626.

Reply:
Please see:
```
	// If there are no nominated pods for the node or if the first run of the
	// predicates fail, we don't run the second pass.
```

Reply:
It corresponds to https://github.com/kubernetes/kubernetes/blob/16d33c49858579fbe13df52c065dbea6627e9aef/pkg/scheduler/core/generic_scheduler.go#L632-L633

Reply:
`alwaysCheckAllPredicates` has nothing to do with preemption logic.

Reply:
podFitsOnNode is called twice by selectVictimsOnNode with alwaysCheckAllPredicates being false.

Reply:
No matter how `podFitsOnNode()` is called, it functions as what it supposed to be.

So I still can't get the point why you think `alwaysCheckAllPredicates` fits here.